src/pages/Account/Admin.tsx

```diff
@@ -5,61 +5,192 @@ import { useUser } from "hooks/useUser";
+    </>
+  );
+};
+
+const VenueInfosComponent: React.FC<VenueDetailsPartProps> = ({ venue }) => {
+  return (
+    <>
+      <Link
+        to={`/v/${createUrlSafeName(venue.name)}`}
+        target="_blank"
+        rel="noopener noreferer"
+      >
+        Preview
+      </Link>
+    </>
+  );
+};
+
+const EventsComponent: React.FC<VenueDetailsPartProps> = ({ venue }) => {
   useFirestoreConnect([
     {
       collection: "venues",
-      where: [["owners", "array-contains", user?.uid]],
+      doc: venue.id,
+      subcollections: [{ collection: "events" }],
+      orderBy: ["start_utc_seconds", "desc"],
+      storeAs: "events",
     },
   ]);
 
+  const events = useSelector((state) => state.firestore.ordered.events);
+  const [showCreateEventModal, setShowCreateEventModal] = useState(false);
+
   return (
-    <WithNavigationBar>
-      <div className="container admin-container">
-        <div className="title">Admin</div>
-        <AuthenticationModal show={!user} onHide={() => {}} showAuth="login" />
-        My Venues
-        {venues?.map((venue) => (
-          <div className="information-card-container admin-item" key={venue.id}>
-            <h4>{venue.name}</h4>
-            <Link to={`/admin/venue/${venue.id}`} className="btn btn-primary">
-              Edit
-            </Link>
-          </div>
+    <>
+      <ul>
+        {events?.map((event) => (
+          <li key={event.id}>
+            {event.name} - {event.description} (£{event.price})
+          </li>
         ))}
-        <div className="centered-flex">
-          <button
-            className="btn btn-primary"
-            onClick={() => push("/admin/venue/creation")}
-          >
-            Create a venue
-          </button>
-        </div>
-        <div className="centered-flex">
-          <button
-            className="btn btn-primary"
-            onClick={() => setShowCreateEventModal(true)}
-          >
-            Create an Event
-          </button>
-        </div>
+      </ul>
+      <div className="centered-flex">
+        <button
+          className="btn btn-primary"
+          onClick={() => setShowCreateEventModal(true)}
+        >
+          Create an Event
+        </button>
       </div>
       <AdminEvent
         show={showCreateEventModal}
         onHide={() => {
           setShowCreateEventModal(false);
         }}
-        venueId={"demo-event-creation"}
+        venueId={venue.id}
       />
+    </>
+  );
+};
+
+const Admin: React.FC = () => {
+  const { user } = useUser();
+  const { venueId } = useParams();
+
+  useFirestoreConnect([
+    {
+      collection: "venues",
+      where: [["owners", "array-contains", user?.uid]],
+    },
+  ]);
+
+  return (
+    <WithNavigationBar>
+      <div className="container admin-container">
+        <AuthenticationModal show={!user} onHide={() => {}} showAuth="login" />
+        <div className="title">Admin</div>
+        <div style={{ flex: "1", flexDirection: "row", display: "flex" }}>
```

Comment:
no need for `display: flex` with `flex: 1`, right?

Reply:
Apparently it is necessary. Doesn't respect the row flow without. Going to disappear tomorrow when we add the styles.